src/Castle.Core/DynamicProxy/ModuleScope.cs

```diff
@@ -81,6 +81,16 @@ public ModuleScope(bool savePhysicalAssembly)
 		{
 		}
 
+		/// <summary>
+		///   Initializes a new instance of the <see cref = "ModuleScope" /> class,
+		///   allowing to specify the <see cref="AssemblyBuilderAccess" /> to be used for generated assemblies.
+		/// </summary>
+		/// <param name = "assemblyBuilderAccess">The desired <see cref="AssemblyBuilderAccess" /> to be used for the generated assemblies.</param>
+		public ModuleScope(AssemblyBuilderAccess assemblyBuilderAccess)
+			: this(assemblyBuilderAccess, disableSignedModule: false, new NamingScope(), DEFAULT_ASSEMBLY_NAME, DEFAULT_FILE_NAME, DEFAULT_ASSEMBLY_NAME, DEFAULT_FILE_NAME)
+		{
+		}
```

Comment:
This new constructor overload doesn't offer any other options for strong-naming and assembly save paths. We can always add more overloads, but for now it seems improbable that users would be much interested in those aspects on newer platforms where `AssemblyBuilderAccess.RunAndCollect` is functioning well... and if you're not after `RunAndCollect`, you have all the preexisting overloads already.

Reply:
Decided to change this back to `internal` for now. We can always make it `public` later if desired.